libobs/util/text-lookup.c

```diff
@@ -71,7 +72,7 @@ static struct text_node *text_node_bychar(struct text_node *node, char ch)
 
 	while (subnode) {
 		if (!dstr_is_empty(&subnode->str) &&
-		    subnode->str.array[0] == ch)
+		    toupper(subnode->str.array[0]) == toupper(ch))
```

Comment:
I'd recommend doing something like `ch = toupper(ch);` only once right after the `struct text_node *subnode = node->first_subnode;` line rather than needlessly calling it every time in the while loop. After that, `toupper(subnode->str.array[0]) == ch` should be fine here. Just a minor nitpick, but it's best to avoid needless extra calls to a function that just gets the same value over and over again for no reason when you could just do it once at the beginning of the function call.

Ideally it'd be nice if we stored all identifier strings in uppercase so we can minimize the toupper() calls overall, but at this point I'm just nitpicking at my own code.

Reply:
> I'd recommend doing something like ch = toupper(ch); only once right after the struct text_node *subnode = node->first_subnode; line rather than needlessly calling it every time in the while loop.
Ideally it'd be nice if we stored all identifier strings in uppercase so we can minimize the toupper() calls overall, but at this point I'm just nitpicking at my own code.

I didn't care about the time to call `toupper`.
As you suggested, I moved `toupper` to the beginning of creating the data structure to avoid calling `toupper` needlessly.